Source/WebCore/page/LocalFrameView.cpp

```diff
@@ -4513,10 +4513,10 @@ void LocalFrameView::updateScrollCorner()
     RenderElement* renderer = nullptr;
     std::unique_ptr<RenderStyle> cornerStyle;
     IntRect cornerRect = scrollCornerRect();
-    
-    if (!cornerRect.isEmpty()) {
+    Document* doc = m_frame->document();
+
+    if (!((doc && doc->documentElement() && doc->documentElement()->renderer() && doc->documentElement()->renderer()->style().usesStandardScrollbarStyle()) || cornerRect.isEmpty())) {
```

Comment:
Why doesn't this use `rootElementForCustomScrollbarPartStyle()`? Also instead of that long string of null checks, consider a lambda.

Reply:
This can't use that function because that's looking for Scrollbar Pseudo instead of ScrollbarCorner pseudo. There's probably some refactoring that can be done surrounding the legacy pseudo code but that's tangential to this change.

As for a lambda do you mean something like below?

```c++
    auto usesStandardScrollbarStyle = [](auto& doc) {
        if (!doc || !doc->documentElement())
            return false;
        if (!doc->documentElement()->renderer())
            return false;

        return doc->documentElement()->renderer()->style().usesStandardScrollbarStyle();
    };

    if (!(usesStandardScrollbarStyle(doc) || cornerRect.isEmpty())) {
```

Reply:
Have pushed up that lambda change